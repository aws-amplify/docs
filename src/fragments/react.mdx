## What we'll build

In this tutorial, we'll use React to build a photo sharing app that lets users upload photos to a shared photo gallery. We'll use a GraphQL API (AWS AppSync) to access data stored in a NoSQL database (Amazon DynamoDB). In addition, we'll demonstrate how to authenticate users, communicate with our API, manage photo uploads, and set up continuous deployment and hosting.

The app includes the following views:

- A login view
- A photo gallery view
- A photo upload view

## Prerequisites

- [Install Node](https://nodejs.org/en/)
- This tutorial assumes that you're familiar with both JavaScript/ES6 and React. If you need to brush up on React, we recommend going through the [official tutorial](https://reactjs.org/tutorial/tutorial.html).