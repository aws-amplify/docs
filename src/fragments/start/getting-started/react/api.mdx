Now that you’ve created and configured a React app and initialized a new Amplify project, you can add a feature. The first feature you will add is an API.

The Amplify CLI supports creating and interacting with two types of API categories: REST and GraphQL.

The API you will be creating in this step is a GraphQL API using AWS AppSync (a managed GraphQL service) and the database will be Amazon DynamoDB (a NoSQL database).

## Create a GraphQL API and database

Add a [GraphQL API](https://docs.aws.amazon.com/appsync/latest/devguide/designing-a-graphql-api.html) to your app and automatically provision a database by running the following command from the root of your application directory:

```bash
amplify add api
```

Accept the **default values** which are highlighted below:

```console
? Please select from one of the below mentioned services:
# GraphQL
? Provide API name:
# myapi
? Choose the default authorization type for the API:
# API Key
? Enter a description for the API key:
# demo
? After how many days from now the API key should expire:
# 7 (or your preferred expiration)
? Do you want to configure advanced settings for the GraphQL API:
# No
? Do you have an annotated GraphQL schema?
# No
? Choose a schema template:
# Single object with fields (e.g., “Todo” with ID, name, description)
? Do you want to edit the schema now?
# Yes
```

The CLI should open this GraphQL schema in your text editor.

__amplify/backend/api/myapi/schema.graphql__

```graphql
type Todo @model {
  id: ID!
  name: String!
  description: String
}
```

The schema generated is for a Todo app. You'll notice a directive on the `Todo` type of `@model`. This directive is part of the [GraphQL transform](/cli/graphql-transformer/model) library of Amplify. 

The GraphQL Transform Library provides custom directives you can use in your schema that allow you to do things like define data models, set up authentication and authorization rules, configure serverless functions as resolvers, and more.

A type decorated with the `@model` directive will scaffold out the database table for the type (Todo table), the schema for CRUD (create, read, update, delete) and list operations, and the GraphQL resolvers needed to make everything work together.

From the command line, press __enter__ to accept the schema and continue to the next steps.

### Deploying the API

To deploy this backend, run the `push` command:

```bash
amplify push
```

```console
? Are you sure you want to continue? Y

# You will be walked through the following questions for GraphQL code generation
? Do you want to generate code for your newly created GraphQL API? Y
? Choose the code generation language target: javascript
? Enter the file name pattern of graphql queries, mutations and subscriptions: src/graphql/**/*.js
? Do you want to generate/update all possible GraphQL operations - queries, mutations and subscriptions? Y
? Enter maximum statement depth [increase from default if your schema is deeply nested]: 2
```

Now the API is live and you can start interacting with it!

The API you have deployed is for a Todo app, including operations for creating, reading, updating, deleting, and listing todos.

Next, run the following command to check Amplify's status:

```bash
amplify status
```

This will give us the current status of the Amplify project, including the current environment, any categories that have been created, and what state those categories are in. It should look similar to this:

```console
Current Environment: dev

| Category | Resource name | Operation | Provider plugin   |
| -------- | ------------- | --------- | ----------------- |
| Api      | myapi         | No Change | awscloudformation |
```

To view the GraphQL API in the AppSync console at any time, run the following command:

```bash
amplify console api
```

To view your entire app in the Amplify console at any time, run the following command:

```bash
amplify console
```

### (Optional) Test your API

To test this out locally, you can run the `mock` command.

> If you'd like to go ahead and connect the front end, you can [jump to the next step](#connect-frontend-to-api).

```bash
amplify mock api
```

*Note:* `amplify mock api` requires Java.

```console
# If you have not already deployed you API, you will be walked through the following steps for GraphQL code generation
? Choose the code generation language target: javascript (or preferred target)
? Enter the file name pattern of graphql queries, mutations and subscriptions: src/graphql/**/*.js
? Do you want to generate/update all possible GraphQL operations - queries, mutations and subscriptions: Yes
? Enter maximum statement depth [increase from default if your schema is deeply nested] 2
```

This will open the GraphiQL explorer on a local port. From the test environment you can try out different operations locally, like queries and mutations.

Try running a couple of mutations locally and then querying for the todos:

```graphql
mutation createTodo {
  createTodo(input: {
    name: "Build an API"
    description: "Build a serverless API with Amplify and GraphQL"
  }) {
    id
    name
    description
  }
}

query listTodos {
  listTodos {
    items {
      id
      description
      name
    }
  }
}
```

## Connect frontend to API

In this section you will create a way to list and create todos from the React application. To do this, you will create a form with a button to create todos as well as a way to fetch and render the list of todos.

Open __src/App.js__ and replace it with the following code:

```javascript
/* src/App.js */
import React, { useEffect, useState } from 'react'
import Amplify, { API, graphqlOperation } from 'aws-amplify'
import { createTodo } from './graphql/mutations'
import { listTodos } from './graphql/queries'

import awsExports from "./aws-exports";
Amplify.configure(awsExports);

const initialState = { name: '', description: '' }

const App = () => {
  const [formState, setFormState] = useState(initialState)
  const [todos, setTodos] = useState([])

  useEffect(() => {
    fetchTodos()
  }, [])

  function setInput(key, value) {
    setFormState({ ...formState, [key]: value })
  }

  async function fetchTodos() {
    try {
      const todoData = await API.graphql(graphqlOperation(listTodos))
      const todos = todoData.data.listTodos.items
      setTodos(todos)
    } catch (err) { console.log('error fetching todos') }
  }

  async function addTodo() {
    try {
      if (!formState.name || !formState.description) return
      const todo = { ...formState }
      setTodos([...todos, todo])
      setFormState(initialState)
      await API.graphql(graphqlOperation(createTodo, {input: todo}))
    } catch (err) {
      console.log('error creating todo:', err)
    }
  }

  return (
    <div style={styles.container}>
      <h2>Amplify Todos</h2>
      <input
        onChange={event => setInput('name', event.target.value)}
        style={styles.input}
        value={formState.name}
        placeholder="Name"
      />
      <input
        onChange={event => setInput('description', event.target.value)}
        style={styles.input}
        value={formState.description}
        placeholder="Description"
      />
      <button style={styles.button} onClick={addTodo}>Create Todo</button>
      {
        todos.map((todo, index) => (
          <div key={todo.id ? todo.id : index} style={styles.todo}>
            <p style={styles.todoName}>{todo.name}</p>
            <p style={styles.todoDescription}>{todo.description}</p>
          </div>
        ))
      }
    </div>
  )
}

const styles = {
  container: { width: 400, margin: '0 auto', display: 'flex', flexDirection: 'column', justifyContent: 'center', padding: 20 },
  todo: {  marginBottom: 15 },
  input: { border: 'none', backgroundColor: '#ddd', marginBottom: 10, padding: 8, fontSize: 18 },
  todoName: { fontSize: 20, fontWeight: 'bold' },
  todoDescription: { marginBottom: 0 },
  button: { backgroundColor: 'black', color: 'white', outline: 'none', fontSize: 18, padding: '12px 0px' }
}

export default App
```

Let's walk through some of the functions:

__useEffect__ - When the component loads, the `useEffect` hook is called and it invokes the `fetchTodos` function.

__fetchTodos__ - Uses the Amplify `API` category to call the AppSync GraphQL API with the `listTodos` query. Once the data is returned, the items array is passed in to the `setTodos` function to update the local state.

__addTodo__ - Uses the Amplify `API` category to call the AppSync GraphQL API with the `createTodo` mutation. A difference between the `listTodos` query and the `createTodo` mutation is that `createTodo` accepts an argument containing the variables needed for the mutation.

## Run locally

Next, run the app and you should see the form rendered to the screen and be able to create and view the list of todos:

```bash
npm start
```
