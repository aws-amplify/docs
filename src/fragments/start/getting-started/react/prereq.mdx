- This tutorial assumes that you're familiar with both JavaScript/ES6 and React. If you need to brush up on React, we recommend going through the [official tutorial](https://reactjs.org/tutorial/tutorial.html).