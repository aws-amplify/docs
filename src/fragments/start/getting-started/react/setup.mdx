## Create a new React App

To get started, first create a new React app, and then install and use the Amplify CLI to start adding backend capabilities to your app.

From your projects directory, run the following command and respond the prompts:

```bash
npm create vite@latest
✔ Project name: react-amplified
✔ Select a framework: › React
✔ Select a variant: › TypeScript
```

This creates a new React TypeScript app in a directory called **react-amplified**. You can then run the following commands to switch into the new **react-amplified** directory, install the project dependencies, and run the app by using the following commands:

```bash
cd react-amplified
npm install
npm run dev
```

This runs a development server and allows you to see the output generated by the build. You can see the running app by navigating to [http://localhost:5173](http://localhost:5173).

## Initialize a new backend

With the app running, it's time to set up Amplify and create the backend services to support the app.

Open a new terminal. From the **react-amplified** directory, run:

```bash
amplify init
```

When you initialize Amplify you'll be prompted for some information about the app, with the option to accept recommended values:

```console
? Enter a name for the project reactamplified
The following configuration will be applied:

?Project information
| Name:  reactamplified
| Environment: dev
| Default editor: Visual Studio Code
| App type: javascript
| Javascript framework: react
| Source Directory Path: src
| Distribution Directory Path: build
| Build Command: npm run-script build
| Start Command: npm run-script start

? Initialize the project with the above configuration? Yes
Using default provider  awscloudformation
? Select the authentication method you want to use: AWS profile

...

? Please choose the profile you want to use default
```

> Amplify CLI will infer the proper configuration based on the type of project Amplify is being initialized in where possible. Here it detected Create React App and provided the proper configuration for type of app, framework, source, distribution, build, and start options.

When you initialize a new Amplify project, a few things happen:

- It creates a top level directory called **amplify** that stores your backend definition. During the tutorial you'll add capabilities such as a GraphQL API and authentication. As you add features, the **amplify** folder will grow with infrastructure-as-code templates that define your backend stack. Infrastructure-as-code is a best practice way to create a replicable backend stack.
- It creates a file called **amplifyconfiguration.json** in the **src** directory that holds all the configuration for the services you create with Amplify. This is how the Amplify client is able to get the necessary information about your backend services.
- It modifies the **.gitignore** file, adding some generated files to the ignore list
- A cloud project is created for you in the AWS Amplify Console that can be accessed by running `amplify console`. The Console provides a list of backend environments, deep links to provisioned resources per Amplify category, status of recent deployments, and instructions on how to promote, clone, pull, and delete backend resources

## Install Amplify Libraries

The `aws-amplify` package is the main library for working with Amplify Libraries in your projects:

```bash
npm install aws-amplify
```

## Set up frontend

Next, configure Amplify so it can interact with backend services.

Open **src/index.js** and add the following code below the last import:

```javascript
import { Amplify } from 'aws-amplify';
import amplifyconfig from './amplifyconfiguration.json';
Amplify.configure(amplifyconfig);
```

And that's all it takes to configure Amplify. As you add or remove categories and make updates to your backend configuration using the Amplify CLI, the configuration in **aws-exports.js** will update automatically.

Now that your React app is set up and Amplify is initialized, you can add an API in the next step.
