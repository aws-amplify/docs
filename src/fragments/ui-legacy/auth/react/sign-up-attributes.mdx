<table>
  <thead>
    <tr>
      <th>Attribute</th>
      <th>Type</th>
      <th>Description</th>
      <th>Default</th>
      <th>Required</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td data-column="Attribute">header</td>
      <td data-column="Type">string</td>
      <td data-column="Description">the component header</td>
      <td data-column="Default">'Create a new account'</td>
      <td data-column="Required">no</td>
    </tr>
    <tr>
      <td data-column="Attribute">signUpFields</td>
      <td data-column="Type">array</td>
      <td data-column="Description">see below</td>
      <td data-column="Default">see below</td>
      <td data-column="Required">no</td>
    </tr>
    <tr>
      <td data-column="Attribute">defaultCountryCode</td>
      <td data-column="Type">string</td>
      <td data-column="Description">
        the preselected value in the country code dropdown
      </td>
      <td data-column="Default">'1'</td>
      <td data-column="Required">no</td>
    </tr>
    <tr>
      <td data-column="Attribute">hideAllDefaults</td>
      <td data-column="Type">boolean</td>
      <td data-column="Description">
        determines whether all default signup fields are to be hidden. This
        works in conjunction with the signUpFields attribute
      </td>
      <td data-column="Default">
        if there is no signUpFields attribute, defaults to false
      </td>
      <td data-column="Required">no</td>
    </tr>
    <tr>
      <td data-column="Attribute">hiddenDefaults</td>
      <td data-column="Type">array</td>
      <td data-column="Description">
        determines whether particular default fields are hidden
      </td>
      <td data-column="Default">
        N/A (possible values include 'username', 'password', 'phone_number', and
        'email')
      </td>
      <td data-column="Required">no</td>
    </tr>
  </tbody>
</table>