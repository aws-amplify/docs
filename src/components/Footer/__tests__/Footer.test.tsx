import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { Footer } from '../index';

describe('Footer', () => {
  const component = <Footer hasTOC={true} />;
  it('should render the Footer component', async () => {
    render(component);
    const footer = await screen.getByRole('contentinfo');
    expect(footer).toBeInTheDocument();
  });

  it('should render component that matches snapshot', () => {
    const { container } = render(component);
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('should link to terms, privacy, and social media accounts', async () => {
    render(component);
    const footerLinks = await screen.getAllByRole('link');
    const linkUrls = {
      Discord: 'https://discord.gg/amplify',
      X: 'https://x.com/AWSAmplify',
      Github: 'https://github.com/aws-amplify',
      'site terms': 'https://aws.amazon.com/terms/',
      'privacy policy': 'https://aws.amazon.com/privacy/'
    };

    for (let i = 0; i < footerLinks.length; i++) {
      expect(footerLinks[i].getAttribute('href')).toBe(
        linkUrls[footerLinks[i].textContent]
      );
    }
  });

  it('should have styling specific to whether TOC is present or not', async () => {
    render(component);
    const footer = await screen.getByRole('contentinfo');
    expect(footer.classList).toContain('footer--toc');
  });
});
