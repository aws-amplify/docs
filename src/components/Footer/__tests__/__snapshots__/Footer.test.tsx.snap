// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render component that matches snapshot 1`] = `"<div><footer class="amplify-flex footer footer--toc"><div class="amplify-flex footer-wrapper"><div class="amplify-flex footer__content"><div class="amplify-flex color-switcher"><span class="amplify-text color-switcher__label" id=":r1:-colorSwitcherLabel">Site color mode</span><div class="amplify-flex amplify-togglebuttongroup" role="group" aria-labelledby=":r1:-colorSwitcherLabel"><button class="amplify-button amplify-field-group__control amplify-button--small amplify-togglebutton" type="button" aria-pressed="false" title="Light mode"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path d="M6.28364 4.70137L4.32 2.73973L2.78182 4.28493L4.73455 6.24657L6.28364 4.70137ZM3.27273 10.9041H0V13.0959H3.27273V10.9041ZM13.0909 0H10.9091V3.23288H13.0909V0ZM21.2182 4.28493L19.68 2.73973L17.7273 4.70137L19.2655 6.24657L21.2182 4.28493ZM17.7164 19.2986L19.6691 21.2712L21.2073 19.726L19.2436 17.7644L17.7164 19.2986ZM20.7273 10.9041V13.0959H24V10.9041H20.7273ZM12 5.42466C8.38909 5.42466 5.45455 8.3726 5.45455 12C5.45455 15.6274 8.38909 18.5753 12 18.5753C15.6109 18.5753 18.5455 15.6274 18.5455 12C18.5455 8.3726 15.6109 5.42466 12 5.42466ZM10.9091 24H13.0909V20.7671H10.9091V24ZM2.78182 19.7151L4.32 21.2603L6.27273 19.2877L4.73455 17.7425L2.78182 19.7151Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">Light mode</span></button><button class="amplify-button amplify-field-group__control amplify-button--small amplify-togglebutton" type="button" aria-pressed="false" title="Dark mode"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path d="M13.4165 0.00754715C6.51178 -0.232618 1 5.29117 1 11.9918C1 18.6203 6.37969 24 13.0082 24C17.4633 24 21.3299 21.5743 23.4074 17.9719C14.3892 17.6717 8.88941 7.84893 13.4165 0.00754715Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">Dark mode</span></button><button class="amplify-button amplify-field-group__control amplify-button--small amplify-togglebutton amplify-togglebutton--pressed" type="button" aria-pressed="true" title="System preferences"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path d="M12 0C5.376 0 0 5.376 0 12C0 18.624 5.376 24 12 24C18.624 24 24 18.624 24 12C24 5.376 18.624 0 12 0ZM10.8 21.516C6.072 20.928 2.4 16.896 2.4 12C2.4 7.104 6.06 3.072 10.8 2.484V21.516ZM13.2 2.484C14.436 2.64 15.6 3.024 16.644 3.6H13.2V2.484ZM13.2 6H19.488C19.788 6.372 20.064 6.78 20.304 7.2H13.2V6ZM13.2 9.6H21.288C21.384 9.996 21.468 10.392 21.516 10.8H13.2V9.6ZM13.2 21.516V20.4H16.644C15.6 20.976 14.436 21.36 13.2 21.516ZM19.488 18H13.2V16.8H20.304C20.064 17.22 19.788 17.628 19.488 18ZM21.288 14.4H13.2V13.2H21.516C21.468 13.608 21.384 14.004 21.288 14.4Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">System preference</span></button></div></div><p><svg aria-label="AWS" style="margin-inline-end: var(--amplify-space-xs);" class="amplify-icon" viewBox="0 0 24 24" aria-hidden="false"><path fill-rule="evenodd" clip-rule="evenodd" d="M6.76344 10.2123C6.76344 10.5077 6.79536 10.7471 6.85124 10.9228C6.9151 11.0984 6.99492 11.2899 7.10667 11.4975C7.14658 11.5613 7.16254 11.6252 7.16254 11.6811C7.16254 11.7609 7.11465 11.8407 7.01088 11.9205L6.50801 12.2558C6.43617 12.3037 6.36433 12.3276 6.30047 12.3276C6.22065 12.3276 6.14083 12.2877 6.06101 12.2159C5.94926 12.0961 5.85347 11.9684 5.77365 11.8407C5.69383 11.705 5.61401 11.5533 5.5262 11.3698C4.9036 12.1041 4.12135 12.4713 3.17945 12.4713C2.50895 12.4713 1.97415 12.2797 1.58302 11.8966C1.1919 11.5134 0.992343 11.0026 0.992343 10.364C0.992343 9.68552 1.23181 9.13475 1.71872 8.71968C2.20563 8.30461 2.85218 8.09707 3.67435 8.09707C3.94574 8.09707 4.22511 8.12102 4.52045 8.16093C4.81579 8.20084 5.11911 8.2647 5.4384 8.33654V7.75384C5.4384 7.1472 5.31069 6.72414 5.06324 6.4767C4.80781 6.22925 4.37677 6.10952 3.76215 6.10952C3.48277 6.10952 3.19542 6.14145 2.90008 6.21329C2.60474 6.28513 2.31738 6.37293 2.038 6.48468C1.91029 6.54055 1.8145 6.57248 1.75863 6.58845C1.70275 6.60441 1.66284 6.61239 1.63092 6.61239C1.51917 6.61239 1.46329 6.53257 1.46329 6.36495V5.97382C1.46329 5.84611 1.47925 5.75032 1.51917 5.69445C1.55908 5.63857 1.63092 5.5827 1.74267 5.52682C2.02204 5.38314 2.35729 5.26341 2.74842 5.16763C3.13954 5.06386 3.55461 5.01596 3.99363 5.01596C4.94351 5.01596 5.63795 5.23148 6.08495 5.66252C6.52397 6.09355 6.74747 6.74809 6.74747 7.62613V10.2123H6.76344ZM3.52268 11.4256C3.78609 11.4256 4.05749 11.3777 4.34485 11.2819C4.6322 11.1862 4.88763 11.0106 5.10315 10.7711C5.23086 10.6194 5.32665 10.4518 5.37454 10.2602C5.42244 10.0687 5.45436 9.83718 5.45436 9.56579V9.23054C5.22288 9.17466 4.97543 9.12677 4.72001 9.09484C4.46458 9.06291 4.21713 9.04695 3.96968 9.04695C3.43488 9.04695 3.04376 9.15072 2.78034 9.36623C2.51693 9.58175 2.38922 9.88507 2.38922 10.2842C2.38922 10.6593 2.485 10.9387 2.68456 11.1303C2.87613 11.3298 3.15551 11.4256 3.52268 11.4256ZM9.93235 12.2877C9.78867 12.2877 9.69288 12.2638 9.62903 12.2079C9.56517 12.16 9.50929 12.0482 9.4614 11.8966L7.5856 5.72638C7.5377 5.56673 7.51376 5.46296 7.51376 5.40709C7.51376 5.27938 7.57762 5.20754 7.70533 5.20754H8.48758C8.63924 5.20754 8.74301 5.23148 8.79888 5.28736C8.86274 5.33525 8.91063 5.447 8.95853 5.59866L10.2995 10.8828L11.5447 5.59866C11.5847 5.43902 11.6325 5.33525 11.6964 5.28736C11.7603 5.23946 11.872 5.20754 12.0157 5.20754H12.6543C12.8059 5.20754 12.9097 5.23148 12.9735 5.28736C13.0374 5.33525 13.0933 5.447 13.1252 5.59866L14.3864 10.9467L15.7673 5.59866C15.8152 5.43902 15.8711 5.33525 15.9269 5.28736C15.9908 5.23946 16.0946 5.20754 16.2382 5.20754H16.9806C17.1083 5.20754 17.1801 5.27139 17.1801 5.40709C17.1801 5.44118 17.1743 5.47527 17.1676 5.51433L17.1642 5.5348C17.1562 5.5827 17.1402 5.64655 17.1083 5.73436L15.1846 11.9046C15.1367 12.0642 15.0808 12.168 15.017 12.2159C14.9531 12.2638 14.8494 12.2957 14.7137 12.2957H14.0272C13.8755 12.2957 13.7718 12.2717 13.7079 12.2159C13.644 12.16 13.5882 12.0562 13.5562 11.8966L12.319 6.74809L11.0898 11.8886C11.0498 12.0482 11.002 12.152 10.9381 12.2079C10.8742 12.2638 10.7625 12.2877 10.6188 12.2877H9.93235ZM20.1894 12.5032C19.7743 12.5032 19.3593 12.4553 18.9602 12.3595C18.561 12.2638 18.2497 12.16 18.0422 12.0403C17.9145 11.9684 17.8267 11.8886 17.7948 11.8168C17.7628 11.7449 17.7469 11.6651 17.7469 11.5933V11.1862C17.7469 11.0185 17.8107 10.9387 17.9305 10.9387C17.9784 10.9387 18.0262 10.9467 18.0741 10.9627C18.1051 10.973 18.146 10.9899 18.1926 11.0093C18.2181 11.0199 18.2454 11.0312 18.2737 11.0425C18.5451 11.1622 18.8404 11.258 19.1517 11.3219C19.471 11.3857 19.7823 11.4176 20.1016 11.4176C20.6045 11.4176 20.9956 11.3298 21.267 11.1542C21.5384 10.9786 21.6821 10.7232 21.6821 10.3959C21.6821 10.1724 21.6102 9.98884 21.4666 9.83718C21.3229 9.68552 21.0515 9.54982 20.6604 9.42211L19.5029 9.06291C18.9202 8.87932 18.4892 8.60793 18.2258 8.24873C17.9624 7.89752 17.8267 7.50639 17.8267 7.09132C17.8267 6.75607 17.8985 6.46073 18.0422 6.2053C18.1859 5.94988 18.3775 5.72638 18.6169 5.55077C18.8564 5.36718 19.1278 5.23148 19.4471 5.1357C19.7664 5.03991 20.1016 5 20.4528 5C20.6284 5 20.812 5.00798 20.9876 5.03193C21.1712 5.05587 21.3388 5.0878 21.5064 5.11973L21.5065 5.11973C21.6661 5.15964 21.8178 5.19955 21.9614 5.24745C22.1051 5.29534 22.2169 5.34323 22.2967 5.39113C22.4084 5.45498 22.4883 5.51884 22.5362 5.59068C22.5841 5.65454 22.608 5.74234 22.608 5.85409V6.22925C22.608 6.39688 22.5441 6.48468 22.4244 6.48468C22.3606 6.48468 22.2568 6.45275 22.1211 6.38889C21.6661 6.18136 21.1552 6.07759 20.5885 6.07759C20.1335 6.07759 19.7743 6.14943 19.5269 6.30109C19.2794 6.45275 19.1517 6.68423 19.1517 7.0115C19.1517 7.235 19.2316 7.42657 19.3912 7.57823C19.5508 7.72989 19.8462 7.88156 20.2692 8.01725L21.4027 8.37645C21.9774 8.56004 22.3925 8.81547 22.6399 9.14273C22.8874 9.47 23.0071 9.84516 23.0071 10.2602C23.0071 10.6035 22.9353 10.9148 22.7996 11.1862C22.6559 11.4576 22.4643 11.697 22.2169 11.8886C21.9694 12.0881 21.6741 12.2318 21.3309 12.3356C20.9717 12.4473 20.5965 12.5032 20.1894 12.5032ZM21.698 16.3825C19.0719 18.3222 15.2564 19.3519 11.9758 19.3519C7.37806 19.3519 3.23533 17.6517 0.106324 14.826C-0.141123 14.6025 0.0823771 14.2992 0.377717 14.4748C3.76215 16.4384 7.93681 17.6278 12.2552 17.6278C15.1686 17.6278 18.3695 17.0211 21.3149 15.7759C21.7539 15.5763 22.1291 16.0633 21.698 16.3825ZM19.7185 15.0336C20.5725 14.9298 22.4563 14.7063 22.7916 15.1373C23.1268 15.5604 22.4164 17.3404 22.0971 18.1306C22.0014 18.3701 22.2089 18.4659 22.4244 18.2823C23.8293 17.1009 24.1964 14.6345 23.9091 14.2753C23.6217 13.924 21.1552 13.6207 19.6546 14.6744C19.4231 14.842 19.463 15.0655 19.7185 15.0336Z" fill="currentColor"></path></svg>Amplify open source software, documentation and community are supported by Amazon Web Services.</p><p> Â© 2024, Amazon Web Services, Inc. and its affiliates.</p><p>All rights reserved. View the <a href="https://aws.amazon.com/terms/" class="footer-link" rel="noopener noreferrer" target="_blank">site terms</a> and <a href="https://aws.amazon.com/privacy/" class="footer-link" rel="noopener noreferrer" target="_blank">privacy policy</a>.</p><p>Flutter and the related logo are trademarks of Google LLC. We are not endorsed by or affiliated with Google LLC.</p></div><ul class="amplify-flex footer__links"><li><a href="https://x.com/AWSAmplify" class="footer-link footer-link--social" rel="noopener noreferrer" target="_blank"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path d="M13.9761 10.1624L22.7186 0H20.6469L13.0558 8.82384L6.99289 0H0L9.16837 13.3432L0 24H2.07179L10.0881 14.6817L16.4911 24H23.4839L13.9761 10.1624ZM2.81829 1.55962H6.00044L20.6479 22.5113H17.4657L2.81829 1.55962Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">X</span></a></li><li><a href="https://discord.gg/amplify" class="footer-link footer-link--social" rel="noopener noreferrer" target="_blank"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" clip-rule="evenodd" d="M4.12482 0H19.8767C21.205 0 22.2865 1.104 22.2865 2.472V24L19.7591 21.72L18.3367 20.376L16.8321 18.948L17.4551 21.168H4.12482C2.7965 21.168 1.71503 20.064 1.71503 18.696V2.472C1.71503 1.104 2.7965 0 4.12482 0ZM14.5634 14.7C14.916 15.156 15.3392 15.672 15.3392 15.672C17.9371 15.588 18.9362 13.848 18.9362 13.848C18.9362 9.984 17.2435 6.852 17.2435 6.852C15.5508 5.556 13.9403 5.592 13.9403 5.592L13.7758 5.784C15.7741 6.408 16.7028 7.308 16.7028 7.308C15.4802 6.624 14.2812 6.288 13.1645 6.156C12.3181 6.06 11.507 6.084 10.79 6.18C10.7284 6.18 10.6758 6.18915 10.6165 6.19945C10.6079 6.20095 10.5991 6.20248 10.5901 6.204C10.1787 6.24 9.17952 6.396 7.92172 6.96C7.48678 7.164 7.22817 7.308 7.22817 7.308C7.22817 7.308 8.20384 6.36 10.3198 5.736L10.2022 5.592C10.2022 5.592 8.59176 5.556 6.89903 6.852C6.89903 6.852 5.20629 9.984 5.20629 13.848C5.20629 13.848 6.19372 15.588 8.7916 15.672C8.7916 15.672 9.22654 15.132 9.57919 14.676C8.08629 14.22 7.52205 13.26 7.52205 13.26C7.52205 13.26 7.6396 13.344 7.85119 13.464C7.86294 13.476 7.8747 13.488 7.89821 13.5C7.91584 13.512 7.93348 13.521 7.95111 13.53C7.96874 13.539 7.98637 13.548 8.00401 13.56C8.29788 13.728 8.59176 13.86 8.86213 13.968C9.34409 14.16 9.92009 14.352 10.5901 14.484C11.4718 14.652 12.5062 14.712 13.6347 14.496C14.1872 14.4 14.7514 14.232 15.3392 13.98C15.7506 13.824 16.2091 13.596 16.691 13.272C16.691 13.272 16.1033 14.256 14.5634 14.7ZM8.67425 11.3997C8.67425 10.6677 9.20323 10.0677 9.87327 10.0677C10.5433 10.0677 11.084 10.6677 11.0723 11.3997C11.0723 12.1317 10.5433 12.7317 9.87327 12.7317C9.21499 12.7317 8.67425 12.1317 8.67425 11.3997ZM12.9649 11.3997C12.9649 10.6677 13.4938 10.0677 14.1639 10.0677C14.8339 10.0677 15.3629 10.6677 15.3629 11.3997C15.3629 12.1317 14.8339 12.7317 14.1639 12.7317C13.5056 12.7317 12.9649 12.1317 12.9649 11.3997Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">Discord</span></a></li><li><a href="https://github.com/aws-amplify" class="footer-link footer-link--social" rel="noopener noreferrer" target="_blank"><svg class="amplify-icon" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" clip-rule="evenodd" d="M12.0092 0C5.39329 0 0.0449829 5.49998 0.0449829 12.3042C0.0449829 17.7432 3.47182 22.3472 8.22576 23.9767C8.82013 24.0991 9.03784 23.7119 9.03784 23.3862C9.03784 23.1009 9.01825 22.1232 9.01825 21.1044C5.6901 21.8379 4.99704 19.6377 4.99704 19.6377C4.46219 18.2117 3.6697 17.8452 3.6697 17.8452C2.5804 17.0914 3.74905 17.0914 3.74905 17.0914C4.95737 17.1729 5.59141 18.3544 5.59141 18.3544C6.66087 20.2284 8.38421 19.6989 9.07752 19.3729C9.17645 18.5784 9.4936 18.0284 9.83033 17.7229C7.17589 17.4377 4.38309 16.3784 4.38309 11.6522C4.38309 10.3077 4.85819 9.20771 5.611 8.35222C5.49223 8.04672 5.07615 6.78347 5.73002 5.09273C5.73002 5.09273 6.74022 4.76673 9.01801 6.35572C9.99321 6.08639 10.9989 5.94938 12.0092 5.94823C13.0194 5.94823 14.0492 6.09097 15.0001 6.35572C17.2781 4.76673 18.2883 5.09273 18.2883 5.09273C18.9422 6.78347 18.5259 8.04672 18.4071 8.35222C19.1798 9.20771 19.6353 10.3077 19.6353 11.6522C19.6353 16.3784 16.8425 17.4172 14.1682 17.7229C14.6041 18.1099 14.9803 18.8432 14.9803 20.0044C14.9803 21.6544 14.9607 22.9787 14.9607 23.3859C14.9607 23.7119 15.1786 24.0991 15.7728 23.9769C20.5267 22.3469 23.9535 17.7432 23.9535 12.3042C23.9731 5.49998 18.6052 0 12.0092 0Z" fill="currentColor"></path></svg><span class="amplify-visually-hidden">Github</span></a></li></ul></div></footer></div>"`;
